helm-charts/splunk-otel-collector/templates/deployment-cluster-receiver.yaml

```diff
@@ -104,6 +104,31 @@ spec:
               mountPath: /splunk-messages
             - mountPath: /conf
               name: collector-configmap
+        - name: patch-log-dirs
+          image: {{ template "splunk-otel-collector.image.initPatchLogDirs" . }}
+          imagePullPolicy: {{ .Values.image.initPatchLogDirs.pullPolicy }}
+          command: ['sh', '-c', '
+          mkdir -p {{ .Values.logsCollection.checkpointPath }};
+          chown -Rv {{ $clusterReceiver.securityContext.runAsUser | default 999 }}:{{ $clusterReceiver.securityContext.runAsGroup | default 999 }} {{ .Values.logsCollection.checkpointPath }};
+          chmod -v g+rwxs {{ .Values.logsCollection.checkpointPath }};
+
+          {{- if .Values.splunkPlatform.sendingQueue.persistentQueue.enabled }}
+          mkdir -p {{ .Values.splunkPlatform.sendingQueue.persistentQueue.storagePath }}/receiver;
+          chown -Rv {{ $clusterReceiver.securityContext.runAsUser | default 999 }}:{{ $clusterReceiver.securityContext.runAsGroup | default 999 }} {{ .Values.splunkPlatform.sendingQueue.persistentQueue.storagePath }}/receiver;
+          chmod -v g+rwxs {{ .Values.splunkPlatform.sendingQueue.persistentQueue.storagePath }}/receiver;
+          setfacl -n -Rm d:m::rx,m::rx,d:g:{{ $clusterReceiver.securityContext.runAsGroup | default 999 }}:rx,g:{{ $clusterReceiver.securityContext.runAsGroup | default 999 }}:rx {{ .Values.splunkPlatform.sendingQueue.persistentQueue.storagePath }}/receiver;
+          {{- end }}']
+          securityContext:
+            runAsUser: 0
```

Comment:
We probably need to incorporate the value for h[ttps://github.com/signalfx/splunk-otel-collector-chart/blob/7d907de5a02537ad01455f20002e0c4fa42d4b4c/helm-charts/splunk-otel-collector/values.yaml#L478 ](clusterReceiver.securityContext) here.

Reply:
Maybe a note in here:
https://github.com/signalfx/splunk-otel-collector-chart/blob/main/docs/advanced-configuration.md#running-the-container-in-non-root-user-mode
?

Reply:
I'd suggest adding documentation primarily in the [Advanced Configuration: Data Persistence](https://github.com/signalfx/splunk-otel-collector-chart/blob/main/docs/advanced-configuration.md#data-persistence) section. Then, include a brief note with a hyperlink in the [Running the Container in Non-Root User Mode](https://github.com/signalfx/splunk-otel-collector-chart/blob/main/docs/advanced-configuration.md#running-the-container-in-non-root-user-mode) section that points to the data persistence section.